src/unix/lwt_io.ml

```diff
@@ -1470,6 +1470,22 @@ let create_temp_dir
   in
   attempt 0
 
+let win32_unlink fn =
+  try Lwt_unix.unlink fn with
+  | Unix.Unix_error (Unix.EACCES, _, _) as e -> (
+    try
+      (* Try removing the read-only attribute *)
+      Lwt_unix.chmod fn 0o666 >>= fun () ->
+      Lwt_unix.unlink fn
+    with
+    | _ -> raise e)
```

Comment:
Maybe it complicates things a bit but is it worth resetting the mod to the original setting in case the second unlink fails?

It does complicate the code to do so (read stats before chmod, catch exceptions of unlink specifically,restore perms before reraising), but at least it'd be more likely to leave the file in its original state.

------------------------------

In addition, I think you should be using `Lwt.catch` to handle exceptions here. (Although I might be missing something so please don't hesitate to let me know if there's a reason that `try`-`with` is appropriate here specifically.)

Reply:
I applied the two suggestions. The original permissions are fetched with `lstat`, if I'm not mistaken that's the correct syscall in our case.
I hadn't thought about using `Lwt.catch`, I converted the code to it.

Reply:
I noticed that you made a subtle change and I'm not sure if it's on purpose: You now reraise the exception from the second `unlink` whereas you previously reraised the original exception.

I don't think one is necessary better than the other (in the specific case of a system programming library for windows), but I'd rather the semantic was not chosen accidentally. Is the change intended or not?